### PR TITLE
User dshboard backend

### PR DESCRIPTION
**Pull request by harshraj22**
Updated user dashboard backend #36 
For reviewers :
* see newComers.md in case you have trouble fetching this branch locally
- [ ] update your login.php (add your own user name and password, database = "cse_hub" , hostname = "localhost")

- [ ] Update your practice_questions_info table (
 +-----------+------------------+---------------------------------+------------------------+---------+
| ques_ID | date_created | successful_submissions | total_submissions | admin |
+-----------+-------------------+---------------------------------+------------------------+----------+
| q1           | NULL             |                      0                |                 0           | super  |
+------------+------------------+---------------------------------+------------------------+---------+

- [ ] update your user_info table
+-------------+-----------------------+--------+
| username | email                  | name |
+--------------+------------------- --+-------+
| user          | user@user.com | user   |
+-------------+----------------------+---------+

* Front end hasn't been updated, so for now go to al.html
* try logging in using usernames that exist and that doesn't 
* try links given and report any vulnerability
* try to directly access dashboard.php ( in this case, dashboard.php doesn't have info regarding whose dashboard to display )

### CODE REVIEW

File: CSE-Hub/user/dashboard.php [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+                        <th>
+                            Successful Submission
+                        </th>
+                        <th>
+                            Date Created     
+                        </th>
+                    </tr>
+                    <?php
+                        
+                        for($i = 0;$i<$newNumOfRows;$i++){
```
Comment (JanhaviDadhania) [inline CSE-Hub/user/dashboard.php]: $newRow = mysqli_fetch_row($newResult);
this line should be added here.

**Review by JanhaviDadhania — COMMENTED** [review]
on dashboard it displays only first row of practice questions info table. 
if multiple rows are present in table then it displays same row multiple times.

**Review by JanhaviDadhania — CHANGES_REQUESTED** [review]
$newRow = mysqli_fetch_row($newResult);
this line should be added inside for loop.

**Comment by harshraj22** [comment]
> $newRow = mysqli_fetch_row($newResult);
> this line should be added inside for loop.

Updated :+1:

**Comment by harshraj22** [comment]
Current features:
* Shows 10 practice questions in the right pannel (has option to sort based on number of successful submissions, date created)
* Each of these is link which displays the actual question (now shows directly .txt file)
* Has a link that shows all the questions, which contains link for each questions (when clicked shows those .txt files).

To do:
* Add user submissions link.

**Comment by DestroyerAlpha** [comment]
Small change:
Error 4040 -> Error 404

**Comment by harshraj22** [comment]
Updated to add user submissions page linked to user  dashboard. Fetches all details directly form user solutions table.
#45 

contains page to show all practice questions #42

File: CSE-Hub/user/dashboard.php [hunk 2]
```diff
@@ -12,17 +12,27 @@
...
+    if(isset($_GET['sortBy'])){
+        $sort = ' ORDER BY '.trim($_GET['sortBy']);
+        if(isset($_GET['order']) && $_GET['order']==1)
+            $order = ' DESC ';
+    }
+    else 
+        $sort = '';
+    $newQuery = "SELECT * FROM practice_questions_info ". $sort . $order." LIMIT 10";
+    
+    echo $newQuery;
```
Comment (JanhaviDadhania) [inline CSE-Hub/user/dashboard.php]: shouldn't this be commented?
line 35

Comment (harshraj22) [reply CSE-Hub/user/dashboard.php]: updated

**Comment by DestroyerAlpha** [comment]
So you have just designed the basic user submissions page right? Because it needs some spice :100: 
Like adding some border to the tables.
Let's do that later.
We can merge this.

**Comment by harshraj22** [comment]
Yeah, this wasn't aimed for frontend.

**Pull request merged**